src/parenttext_pipeline/cli.py

```diff
@@ -1,14 +1,45 @@
+import argparse
 import runpy
 
-from parenttext_pipeline.pipelines import Config, run
+from parenttext_pipeline.configs import Config
+import parenttext_pipeline.compile_flows
+import parenttext_pipeline.pull_data
+
+
+PIPELINE_VERSION = "1.0.0"
+
+
+OPERATIONS_MAP = {
+    "pull_data" : parenttext_pipeline.pull_data.run,
+    "compile_flows" : parenttext_pipeline.compile_flows.run,
+}
 
 
 class ConfigError(Exception):
     pass
 
 
 def init():
-    run(load_config())
+    parser = argparse.ArgumentParser(
+        description="Run a pipeline of operations."
+    )
+    parser.add_argument(
+        "operations",
+        nargs="+",
+        help=(
+            "Sequence of operations to perform. Valid choices: pull_data, compile_flows."
+        ),
+    )
+    args = parser.parse_args()
+
+    config = load_config()
+
+    config_pipeline_version = config.meta["pipeline_version"]
+    if config_pipeline_version != PIPELINE_VERSION:
```

Comment:
Pipeline version should ideally be guaranteed to be in sync with a github tag? Not sure how to do that.

Also, this is probably a bit too strict. Maybe we could have versioning so that if only the last number changes, we're assuming backward-compatibility. So major and minor must be the same, and we must have revision (config) < revision (pipeline)?

Reply:
Minor releases should be backwards-compatible as well. If the config file states v1.2, but the pipeline is v1.1 it should still be able to run to completion, although this may not be what the user intended, so a warning may be more appropriate than stopping execution.

I have tried using setuptools-scm to version the goals api based on tags, so you may want to try that also.

Reply:
Thanks, I'll look into that.